src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientFactory.cs

```diff
@@ -2,21 +2,32 @@
 // Licensed under the MIT license.
 //
 using Azure.Core;
-using Azure.Data.AppConfiguration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.ConfigurationClients;
 using System;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
 {
     internal class ConfigurationClientFactory : IConfigurationClientFactory
     {
-        public ConfigurationClient CreateConfigurationClient(string connectionString, ConfigurationClientOptions clientOptions)
+        public IConfigurationClient CreateConfigurationClient(string connectionString, AzureAppConfigurationOptions options)
         {
-            return new ConfigurationClient(connectionString, clientOptions); 
+            return new FailOverSupportedConfigurationClient(connectionString, options);
         }
 
-        public ConfigurationClient CreateConfigurationClient(Uri endpoint, TokenCredential credential, ConfigurationClientOptions clientOptions)
+        public IConfigurationClient CreateConfigurationClient(IEnumerable<Uri> endpoints, TokenCredential credential, AzureAppConfigurationOptions options)
         {
-            return new ConfigurationClient(endpoint, credential, clientOptions);
+            if (endpoints == null || endpoints.Count() < 1)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
```

Comment:
Is the added parameter check necessary? it should never get this far if any of those is the case. It's like in the other overload, you don't check the connectingString is not null.

Reply:
It is not valid anymore since we got rid of this class.